src/com/google/cloud/datastore/node/DatastoreDAO.js

```diff
@@ -0,0 +1,267 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+foam.CLASS({
+  package: 'com.google.cloud.datastore.node',
+  name: 'SelectData',
+
+  documentation: function() {/*
+                               State passed around by intermediate callbacks
+                               during a select() in progress. These data must
+                               be retained to notify the sink, send the
+                               correct payload to subsequent API calls, and
+                               return results in the Promise.
+                              */},
+
+  properties: [
+    {
+      name: 'sink'
+    },
+    {
+      name: 'requestPayload'
+    },
+    {
+      class: 'Array',
+      name: 'results',
+    }
+  ]
+});
+
+foam.CLASS({
+  package: 'com.google.cloud.datastore.node',
+  name: 'DatastoreDAO',
+  extends: 'foam.dao.AbstractDAO',
+
+  documentation: function() {/*
+                               DAO implementation that speaks of Cloud
```

Comment:
Less creative documentation comment?

DAO implementation for the Google Cloud Datastore v1 REST API.